### ISSUE TITLE
1.8.16: Blank FILE_PATTERNS => no files processed

### ISSUE THREAD

**Issue by balducci** [doxygen/doxygen#7190]
hello

apologies if I am missing some trivial point here

looks like there is a small problem/inconsistency with recently
released doxygen-1.8.16

The problem relates to the FILE_PATTERNS tag
Both in the manual and in the comment inside a generated configuration
file it is said:

    ----8<----
    # If left blank the following patterns are tested:*.c, *.cc,
    # *.cxx, *.cpp, *.c++, *.java, *.ii, *.ixx, *.ipp, *.i++,
    # *.inl, *.idl, *.ddl, *.odl, *.h, *.hh, *.hxx, *.hpp,
    # *.h++, *.cs, *.d, *.php, *.php4, *.php5, *.phtml, *.inc,
    # *.m, *.markdown, *.md, *.mm, *.dox, *.py, *.pyw, *.f90,
    # *.f95, *.f03, *.f08, *.f, *.for, *.tcl, *.vhd, *.vhdl,
    # *.ucf, *.qsf and *.ice.
    ---->8----

and, actually, 1.8.15 behaves accordingly.

But 1.8.16 behaves differently: AFAICS, if I leave FILE_PATTERNS
blank, as in:

    ----8<----
    FILE_PATTERNS =
    ---->8----

then C sources in a directory specified with INPUT are not
processed. Instead, to have them recognized and processed I have to:

=> either *comment* FILE_PATTERNS, like in:

    ----8<----
    # FILE_PATTERNS =
    ---->8----

=> or explicitly define extensions, like in:

    ----8<----
    FILE_PATTERNS = *.c
    ---->8----

In summary, manual and comment say that a *blank* FILE_PATTERNS should
trigger default glob patterns: this is true for 1.8.15, but not for
1.8.16 (where a blank FILE_PATTERNS causes, apparently, no recognition
whatsoever of any file extension and thus no file processing).

If this is intentional, then manual and comments should be updated;
otherwise, there seems to be a problem with doxygen-1.8.16 code

I enclose a small tar which demonstrates the problem (run the doit
script)

thanks a lot for your valuable work
ciao
-gabriele

**Comment by albert-github** [comment]
You wrote:
> I enclose a small tar which demonstrates the problem (run the doit script)

But I don't see the file.

**Comment by balducci** [comment]
here is the test case:

[demo.tar.gz](https://github.com/doxygen/doxygen/files/3496749/demo.tar.gz)

**Comment by albert-github** [comment]
I've just pushed a proposed patch, pull request #7193

**Comment by balducci** [comment]
problem fixed
thanks a lot

ciao
-gabriele

**Comment by albert-github** [comment]
Code has been integrated in master on github (please don't close the issue as this will be done at the moment of a release).

**Comment by SSteve** [comment]
This should resolve https://github.com/opencv/opencv/issues/15976

**Comment by doxygen** [comment]
This issue was previously marked 'fixed but not released',
which means it should be fixed in doxygen version 1.8.17.
Please verify if this is indeed the case. Reopen the
issue if you think it is not fixed and please include any additional information
that you think can be relevant (preferably in the form of a self-contained example).
